test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperScopeManagerTest.cs

```diff
@@ -17,17 +18,168 @@ public class TagHelperScopeManagerTest
             async () => await Task.FromResult(result: true);
 
         [Fact]
+        public void Begin_DoesNotRequireParentExecutionContext()
+        {
+            // Arrange & Act
+            var scopeManager = new TagHelperScopeManager();
+
+            // Act
+            var executionContext = scopeManager.Begin(
+                parentExecutionContext: null,
+                tagName: "p",
+                uniqueId: string.Empty,
+                executeChildContentAsync: DefaultExecuteChildContentAsync,
+                startWritingScope: DefaultStartWritingScope,
+                endWritingScope: DefaultEndWritingScope);
+            executionContext.Items["test-entry"] = 1234;
+
+            // Assert
+            var executionContextItem = Assert.Single(executionContext.Items);
+            Assert.Equal("test-entry", (string)executionContextItem.Key, StringComparer.Ordinal);
+            Assert.Equal(1234, executionContextItem.Value);
+        }
+
+        [Fact]
+        public void Begin_ReturnedExecutionContext_ItemsAreRetrievedFromParentExecutionContext()
+        {
+            // Arrange
+            var scopeManager = new TagHelperScopeManager();
+            var parentExecutionContext = new TagHelperExecutionContext("p");
+            parentExecutionContext.Items["test-entry"] = 1234;
+
+            // Act
+            var executionContext = scopeManager.Begin(
+                parentExecutionContext,
+                tagName: "p",
+                uniqueId: string.Empty,
+                executeChildContentAsync: DefaultExecuteChildContentAsync,
+                startWritingScope: DefaultStartWritingScope,
+                endWritingScope: DefaultEndWritingScope);
+
+            // Assert
+            var executionContextItem = Assert.Single(executionContext.Items);
+            Assert.Equal("test-entry", (string)executionContextItem.Key, StringComparer.Ordinal);
+            Assert.Equal(1234, executionContextItem.Value);
+        }
+
+        [Fact]
+        public void Begin_ReturnedExecutionContext_ComplexObjectItemEntriesAreVisibleToParent()
```

Comment:
name is too specific; much closer to `Begin_DoesShallowCopyOfParentItems`
- if you want to be almost this specific `Begin_SetsUpItems_SoParentSeesChangesChildrenMakeToExistingItems` -- expect few dictionaries within dictionaries but lots of complex objects in `Items` where child changes a property value